File: ios/src/main/java/y2k/joyreactor/PostCell.java
package y2k.joyreactor;

import org.robovm.apple.uikit.*;
import org.robovm.objc.annotation.CustomClass;
import org.robovm.objc.annotation.IBAction;
import org.robovm.objc.annotation.IBOutlet;
import y2k.joyreactor.platform.StoryboardNavigation;
import y2k.joyreactor.presenters.PostListPresenter;

/**
 * Created by y2k on 10/11/15.
 */
@CustomClass("PostCell")
public class PostCell extends UITableViewCell {

    @IBOutlet
    UILabel replyCountView;

    @IBOutlet
    UILabel ratingView;

    @IBOutlet
    UIButton playButton;

    @IBOutlet
    UIButton rateButton;

    private PostListPresenter presenter;
    private Post post;

    @IBAction
    void rate() {
        UIAlertController alert = new UIAlertController();
        alert.addAction(new UIAlertAction(Translator.get("Like"), UIAlertActionStyle.Default, s -> {
            // TODO:
        }));
        alert.addAction(new UIAlertAction(Translator.get("Dislike"), UIAlertActionStyle.Destructive, s -> {
            // TODO:
        }));
        alert.addAction(new UIAlertAction(Translator.get("Cancel"), UIAlertActionStyle.Cancel, null));
        StoryboardNavigation.getNavigationController().presentViewController(alert, true, null);
    }

    @IBAction
    void clicked() {
        presenter.postClicked(post);
    }

    @IBAction
    void play() {
        presenter.playClicked(post);
    }

    public void update(PostListPresenter presenter, Post post) {
        this.presenter = presenter;
        this.post = post;

        replyCountView.setText("" + post.commentCount);
        ratingView.setText("" + post.rating);

        playButton.setHidden(post.image == null);
        playButton.setTitle(getPlayButtonTitle(), UIControlState.Normal);
    }

    private String getPlayButtonTitle() {
        return post.image != null && post.image.isAnimated()
                ? Translator.get("Play") : Translator.get("View");
    }
}

File: ios/src/main/kotlin/y2k/joyreactor/PostCell.kt
package y2k.joyreactor

import org.robovm.apple.uikit.*
import org.robovm.objc.annotation.CustomClass
import org.robovm.objc.annotation.IBAction
import org.robovm.objc.annotation.IBOutlet
import y2k.joyreactor.common.translation
import y2k.joyreactor.platform.StoryboardNavigation
import y2k.joyreactor.presenters.PostListPresenter

/**
 * Created by y2k on 10/11/15.
 */
@CustomClass("PostCell")
class PostCell : UITableViewCell() {

    @IBOutlet lateinit var replyCountView: UILabel
    @IBOutlet lateinit var ratingView: UILabel
    @IBOutlet lateinit var playButton: UIButton
    @IBOutlet lateinit var rateButton: UIButton

    private lateinit var presenter: PostListPresenter
    private lateinit var post: Post

    @IBAction
    internal fun rate() {
        val alert = UIAlertController()
        alert.addAction(UIAlertAction("Like".translation, UIAlertActionStyle.Default) {
            // TODO:
        })
        alert.addAction(UIAlertAction("Dislike".translation, UIAlertActionStyle.Destructive) {
            // TODO:
        })
        alert.addAction(UIAlertAction("Cancel".translation, UIAlertActionStyle.Cancel, null))
        StoryboardNavigation.getNavigationController().presentViewController(alert, true, null)
    }

    @IBAction
    internal fun clicked() {
        presenter.postClicked(post)
    }

    @IBAction
    internal fun play() {
        presenter.playClicked(post)
    }

    fun update(presenter: PostListPresenter, post: Post) {
        this.presenter = presenter
        this.post = post

        replyCountView.text = "" + post.commentCount
        ratingView.text = "" + post.rating

        playButton.isHidden = post.image == null
        playButton.setTitle(playButtonTitle, UIControlState.Normal)
    }

    private val playButtonTitle: String
        get() = if (post.image?.isAnimated ?: false) "Play".translation else "View".translation
}